lib/collection/src/shards/replica_set/update.rs

```diff
@@ -363,12 +363,30 @@ impl ShardReplicaSet {
                 );
             }
 
+            // If there is at least one full-complete operation, we can't ignore non-transient errors (4xx)
+            // And we must deactivate failed replicas to ensure consistency
```

Comment:
I don't fully understand why this is the case.

Isn't exactly the same true for all acknowledged updates, because we can assume they'll be applied (eventually) as well?

Could you provide an example which was problematic and has improved now, and potentially a counter example on why it isn't necessary for just acknowledged updates?

Reply:
![image](https://github.com/user-attachments/assets/df8cfbb0-7184-4d18-bb1b-62d0509ea88b)

https://github.com/qdrant/qdrant/blob/5b578c4f34188f0474f901e49d4726213596433d/lib/collection/src/shards/forward_proxy_shard.rs#L243